apsuite/commisslib/measure_tune_bpm.py

```
@@ -0,0 +1,294 @@
+
+    def _create_devices(self):
+        self.devices['currinfo'] = CurrInfoBO()
+        self.devices['bobpms'] = FamBPMs(FamBPMs.DEVICES.BO)
+        self.devices['event'] = Event('DigBO')
+        self.devices['evg'] = EVG()
+        self.devices['trigbpm'] = Trigger('BO-Fam:TI-BPM')
+        self.devices['rfgen'] = RFGen()
+
+    def configure_bpms(self):
+        """."""
+        prms = self.params
+        bobpms = self.devices['bobpms']
+        trigbpm = self.devices['trigbpm']
+
+        bobpms.mturn_config_acquisition(
+            nr_points_after=prms.nr_points_after,
+            nr_points_before=prms.nr_points_before,
+            acq_rate='TbT', repeat=False, external=True)
+        bobpms.mturn_reset_flags()
+        trigbpm.source = prms.trigger_source
+        trigbpm.nr_pulses = prms.nr_pulses
+
+    def get_orbit_data(self, injection=False, external_trigger=False):
+        """Get orbit data from BPMs in TbT acquisition rate
+        BPMs must be configured to listen DigBO event and the DigBO
+        event must be in Injection mode.
+        If injection is True, then injection is turned on before the measure.
+        If external_trigger is True, the event will listen a external trigger.
+        """
+        prms = self.params
+        bobpms = self.devices['bobpms']
+        trigbpm = self.devices['trigbpm']
+
+        self.delay0 = trigbpm.delay
+        trigbpm.delay = self.delay0 + prms.extra_delay
+        self.devices['event'].mode = prms.trigger_source_mode
+
+        # Inject and start acquisition
+        bobpms.mturn_reset_flags()
+        if external_trigger:
+            self.devices['event'].cmd_external_trigger()
+        if injection:
+            self.devices['evg'].cmd_turn_on_injection()
+        ret = bobpms.mturn_wait_update_flags(timeout=prms.bpms_timeout)
+        if ret != 0:
+            print(f'Problem waiting BPMs update. Error code: {ret:d}')
+            trigbpm.delay = self.delay0
+            return dict()
```

If the measurement succeeds, nothing is returned, so why return an empty dict if the measurement fails? A print reporting the errors is already raised.

---

The idea was to get out of the method, but really, return an empty dict() isn't consistent. 
I remove this line and now it stores the data in self.data in any case, but without orbit values if the measure fails.

---

I am thinking that better than that is to raise an exception instead of a print and set an empty dict in self.data. There has no reason to return a bool in this method.